geode-core/src/main/java/org/apache/geode/cache/query/internal/index/IndexManager.java

```diff
@@ -1392,6 +1392,19 @@ public String getCanonicalizedIteratorName(String definition) {
     return ((String) (this.canonicalizedIteratorNameMap.get(definition)));
   }
 
+  /**
+   * Update all indexes with maintenance status
+   *
+   */
+  public void updateIndexMaintenanceStatus(boolean ongoing) {
+    Collection<Index> allindexes = getIndexes();
+    for (Index index : allindexes) {
+      if (index.isMaintenanceOngoing() != ongoing) {
```

Comment:
is this check necessary?

Reply:
Yes, since this method is used for updating maintenance status (ongoing/finished) of all indexes.
We can have scenarios, that we create index, then insert data. Maintenance of index is started. After insert is finished, we can create new index, and this action will be performed synchrones.
And for this index status will be finished, while for first is ongoing.